substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -487,35 +487,62 @@ where
 			"fatp::ready_at_light"
 		);
 
-		let Ok(block_number) = self.api.resolve_block_number(at) else {
+		// Return an empty ready transactions set if we can not find the number of `at` block hash.
+		let Ok(at_number) = self.api.resolve_block_number(at) else {
 			return Box::new(std::iter::empty())
 		};
 
-		let best_result = {
-			api.tree_route(self.enactment_state.lock().recent_finalized_block(), at).map(
-				|tree_route| {
-					if let Some((index, view)) =
-						tree_route.enacted().iter().enumerate().rev().skip(1).find_map(|(i, b)| {
-							self.view_store.get_view_at(b.hash, true).map(|(view, _)| (i, view))
-						}) {
-						let e = tree_route.enacted()[index..].to_vec();
-						(TreeRoute::new(e, 0).ok(), Some(view))
-					} else {
-						(None, None)
-					}
-				},
-			)
+		let mut to_process = Vec::new();
```

Comment:
nit, maybe:
```suggestion
		let mut enacted_blocks = Vec::new();
```

Reply:
Used this naming here: [d11101e](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/d11101e3360100c5a95b546a8d020b3758865b97)